lib/src/rollout_evaluator.dart

```diff
@@ -3,347 +3,1008 @@ import 'dart:convert';
+        _ensureComparisonValue(userCondition.stringArrayValue);
+
+    var matched = false;
+    for (final value in comparisonValues) {
+      if (_ensureComparisonValue(value).isEmpty) {
+        continue;
+      }
+      try {
+        matched = _parseVersion(value.trim()) == userVersion || matched;
+      } catch (e) {
+        return false;
+      }
+    }
+    return negateSemverIsOneOf != matched;
+  }
+
+  bool _evaluateContainsAnyOf(bool negateContainsAnyOf,
+      UserCondition userCondition, String userAttributeValue) {
+    List<String> comparisonValues =
+        _ensureComparisonValue<List<String>>(userCondition.stringArrayValue);
+
+    for (String containsValue in comparisonValues) {
+      if (userAttributeValue.contains(_ensureComparisonValue(containsValue))) {
+        return !negateContainsAnyOf;
+      }
+    }
+    return negateContainsAnyOf;
+  }
+
+  bool _evaluateSegmentCondition(
+      SegmentCondition segmentCondition,
+      EvaluationContext evaluationContext,
+      String? configSalt,
+      List<Segment> segments,
+      EvaluateLogger? evaluateLogger) {
+    int segmentIndex = segmentCondition.segmentIndex;
+    Segment? segment;
+    if (segmentIndex < segments.length) {
+      segment = segments[segmentIndex];
+    }
+    evaluateLogger?.append(
+        EvaluateLogger.formatSegmentFlagCondition(segmentCondition, segment));
+
+    if (evaluationContext.user == null) {
+      if (!evaluationContext.isUserMissing) {
+        evaluationContext.isUserMissing = true;
+        _logger.warning(3001,
+            "Cannot evaluate targeting rules and % options for setting '${evaluationContext.key}' (User Object is missing). You should pass a User Object to the evaluation methods like `getValue()` in order to make targeting work properly. Read more: https://configcat.com/docs/advanced/user-object/");
+      }
+      throw RolloutEvaluatorException(userObjectIsMissing);
+    }
+
+    if (segment == null) {
+      throw ArgumentError("Segment reference is invalid.");
+    }
+    String? segmentName = segment.name;
+    if (segmentName == null || segmentName.isEmpty) {
+      throw ArgumentError("Segment name is missing.");
+    }
+    evaluateLogger?.logSegmentEvaluationStart(segmentName);
+
+    bool result;
+    try {
+      bool segmentRulesResult = _evaluateConditions(segment.segmentRules, null,
+          evaluationContext, configSalt, segmentName, segments, evaluateLogger);
+
+      SegmentComparator segmentComparator =
+          SegmentComparator.tryFrom(segmentCondition.segmentComparator) ??
+              (() => throw ArgumentError(comparisonOperatorIsInvalid))();
+
+      switch (segmentComparator) {
+        case SegmentComparator.isInSegment:
+          result = segmentRulesResult;
+          break;
+        case SegmentComparator.isNotInSegment:
+          result = !segmentRulesResult;
+          break;
+        default:
+          throw ArgumentError(comparisonOperatorIsInvalid);
```

Comment:
Not needed any more because of line 840.

```suggestion
```

Reply:
Line 840 handles the problem if the segmentCondition.segmentComparator contains an invalid value. This handles the case if the comparator is not managed by the switch. At the moment, it is unnecessary, but it is a nice fail-safe for later.
I would leave it as it is.